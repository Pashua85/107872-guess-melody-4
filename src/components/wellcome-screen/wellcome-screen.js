import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import ActionCreator from '../../store/action-creator/action-creator';
import {getMistakesLimit} from '../../store/reducers/gameReducer/selectors';

const WellcomeScreen = (props) => {
  const {mistakesLimit, onStartClick} = props;
  return <section className="welcome">
    <div className="welcome__logo">
      <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
    </div>
    <Link to="/dev-artist">
      <button
        className="welcome__button"
        onClick={onStartClick}
      >
        <span className="visually-hidden">Начать игру</span>
      </button>
    </Link>
    <h2 className="welcome__rules-title">Правила игры</h2>
    <p className="welcome__text">Правила просты:</p>
    <ul className="welcome__rules-list">
      <li>Нужно ответить на все вопросы.</li>
      <li>Можно допустить {mistakesLimit} ошибки.</li>
    </ul>
    <p className="welcome__text">Удачи!</p>
  </section>;
};

WellcomeScreen.propTypes = {
  mistakesLimit: PropTypes.number.isRequired,
  onStartClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  mistakesLimit: getMistakesLimit(state)
});

const mapDispatchToProps = (dispatch) => ({
  onStartClick: () => (
    dispatch(ActionCreator.increaseStepAction())
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(WellcomeScreen);
export {WellcomeScreen};
