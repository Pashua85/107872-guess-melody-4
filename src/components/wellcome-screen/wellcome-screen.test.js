import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {WellcomeScreen} from './wellcome-screen';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`WellcomeScreen`, () => {
  it(`should render correctly with 3 errors`, () => {
    const wrapper = shallow(
        <MemoryRouter>
          <WellcomeScreen
            mistakesLimit={3}
            onStartClick={() => {}}
          />
        </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it(`should render correctly with 2 errors`, () => {
    const wrapper = shallow(
        <MemoryRouter>
          <WellcomeScreen
            mistakesLimit={2}
            onStartClick={() => {}}
          />
        </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
