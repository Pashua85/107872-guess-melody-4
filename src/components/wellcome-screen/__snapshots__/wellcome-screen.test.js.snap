// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WellcomeScreen should render correctly with 2 errors 1`] = `ShallowWrapper {}`;

exports[`WellcomeScreen should render correctly with 3 errors 1`] = `ShallowWrapper {}`;
